src/main.cpp

```diff
@@ -94,14 +111,23 @@ static void handleGeofencing() {
     int withinGeofenceIndex = isWithinGeofence(gnssData.latitude, gnssData.longitude);
 
     if (withinGeofenceIndex >= 0) {
-        static bool ledState = false;
-        digitalWrite(LED_BUILTIN, ledState ? HIGH : LOW);
-        ledState = !ledState;
+        // Check if altitude is within the geofence's altitude range
+        if (isWithinGeofenceAltitude(withinGeofenceIndex, sensorData.baroAltitude)) {
+            static bool ledState = false;
+            digitalWrite(LED_BUILTIN, ledState ? HIGH : LOW);
+            ledState = !ledState;
+            servoOpen(SERVO_CONTROL_PIN);
+        }
     }
     
     if (DEBUG && (withinGeofenceIndex >= 0)) {
         Serial.print("Within geofence index: ");
         Serial.println(withinGeofenceIndex);
+        if (isWithinGeofenceAltitude(withinGeofenceIndex, gnssData.altitude)) {
+            Serial.println("Altitude within geofence range");
+        } else {
+            Serial.println("Altitude outside geofence range");
+        }
```

Comment:
would you be able to merge the logging with the doing. its kinda strange that its checking everything twice